src/textual/widgets/_data_table.py

```diff
@@ -712,29 +1097,99 @@ def _refresh_region(self, region: Region) -> None:
         region = region.translate(-self.scroll_offset)
         self.refresh(region)
 
+    def is_valid_row_index(self, row_index: int) -> bool:
+        """Return a boolean indicating whether the row_index is within table bounds.
+
+        Args:
+            row_index: The row index to check.
+
+        Returns:
+            True if the row index is within the bounds of the table.
+        """
+        return 0 <= row_index < len(self.rows)
+
+    def is_valid_column_index(self, column_index: int) -> bool:
+        """Return a boolean indicating whether the column_index is within table bounds.
+
+        Args:
+            column_index: The column index to check.
+
+        Returns:
+            True if the column index is within the bounds of the table.
+        """
+        return 0 <= column_index < len(self.columns)
+
+    def is_valid_coordinate(self, coordinate: Coordinate) -> bool:
+        """Return a boolean indicating whether the given coordinate is within table
+        bounds.
```

Comment:
May want reformatting to confirm with [our docstring guidelines](https://github.com/Textualize/textual/wiki/Documentation-standards#class).